### PR TITLE
Add custom header parameters in core.

### PR DESCRIPTION
**Pull request by SpectralDragon**
Now there is the possibility to set custom headers using the key `header_params`. I refused to subclass and now all the action takes place in the nucleus `OAuth2AuthRequest`. Thank you in advance :)

Excuse me for my broken English :'D

### CODE REVIEW

**Comment by p2** [comment]
Yeah, I guess this makes sense. Can you rename to `authHeaders` and the dictionary key simply to `headers` and fix the whitespaces (should be tabs, not spaces, and no new newlines inserted). And add yourself to `CONTRIBUTORS.md`!

**Comment by SpectralDragon** [comment]
@p2 I hope of those changes will be enough :)

File: Sources/Base/OAuth2.swift [hunk 1]
```diff
@@ -109,9 +109,9 @@ public class OAuth2: OAuth2Base {
 	}
     
     /// Contains parameters header. Settings key `header_params`
-    public var header: OAuth2StringDict? {
-        get { return clientConfig.headerParams }
-        set { clientConfig.headerParams = newValue }
+    public var headers: OAuth2Headers? {
```
Comment (p2) [inline Sources/Base/OAuth2.swift]: Should be called `authHeaders`. Use tabs for indentation.

File: Sources/Base/OAuth2.swift [hunk 2]
```diff
@@ -429,7 +428,7 @@ public class OAuth2: OAuth2Base {
 		do {
 			let post = try tokenRequestForTokenRefresh(params: params).asURLRequestFor(self)
 			logger?.debug("OAuth2", msg: "Using refresh token to receive access token from \(post.URL?.description ?? "nil")")
-
+
```
Comment (p2) [inline Sources/Base/OAuth2.swift]: Undo this whitespace change.

File: Sources/Base/OAuth2ClientConfig.swift [hunk 3]
```diff
@@ -110,9 +110,8 @@ public class OAuth2ClientConfig {
 		if let assume = settings["token_assume_unexpired"] as? Bool {
 			accessTokenAssumeUnexpired = assume
 		}
-        
-        if let params = settings["header_params"] as? OAuth2StringDict {
-            headerParams = params
+        if let params = settings["header_params"] as? OAuth2Headers {
```
Comment (p2) [inline Sources/Base/OAuth2ClientConfig.swift]: Should be called `settings["headers"]`

**Comment by p2** [comment]
Almost. And please us tabs! Looking at the diff on Github you can see where you're using spaces.

**Comment by SpectralDragon** [comment]
@p2 i don't use spaces Q_Q I use tabs!

**Comment by p2** [comment]
Hmmm, look here, all your changes use spaces. 🤔

https://github.com/p2/OAuth2/pull/128/files

**Comment by SpectralDragon** [comment]
@p2 i use tabs with parameters equal 4 spaces

**Comment by p2** [comment]
Sure, but the files should contain tabs, not spaces. ;)

**Comment by SpectralDragon** [comment]
Hmm, maybe you talked about this?
image

**Comment by p2** [comment]
Exactly. Please take a look here: https://github.com/p2/OAuth2/pull/128/files and see all the whitespace changes that are unneccessary and fix them.

**Comment by SpectralDragon** [comment]
@p2 Wow, this was amazing experience :"D

**Comment by p2** [comment]
Haha, I went through some of these myself. :)

Some spaces left, but I'll accept and fix it myself later. Thanks for the PR!

**Pull request merged**